src/core/ext/transport/chttp2/transport/hpack_parser.c

```diff
@@ -1646,11 +1646,14 @@ static void force_client_rst_stream(grpc_exec_ctx *exec_ctx, void *sp,
   grpc_chttp2_stream *s = sp;
   grpc_chttp2_transport *t = s->t;
   if (!s->write_closed) {
+    grpc_chttp2_mark_stream_closed(exec_ctx, t, s, true, false, false,
```

Comment:
This should not be there. I had kept it originally, but since this will be run in a combiner, there should not be any race